### DISCUSSION TITLE
SCRIPT: save value of a key that is in an external JSON MQTT topic in a script variable and display it

### DISCUSSION THREAD

**Discussion by HaraldNoack** [arendst/Tasmota#17787] [Q&A]
Hello,

I want to display some external MQTT-info on a i2c display connected to a esp32-tasmota.
For this I need to subscribe the following topic on the MQTT broker and extract the values of the needed keys out of the JSON.
Then I have to save the key-values in variables and display them.

MQTT-topic:
tele/Tasmota_Pressure_Sensor/SENSOR

JSON-data-example:
{"Time":"2023-01-24T17:31:07","BME280":{"Temperature":19.1,"Humidity":69.9,"DewPoint":13.4,"Pressure":990.1},"Fan":{"PWM":13418.00,"RPM":539.98},"SDP810":{"DP_Pa":-3.533,"DP_mmH2O":-0.360,"TEMP":21.225},"PressureUnit":"hPa","TempUnit":"C"}

Script:
###################################################################

D
g:PWM=0
g:DP=0.0
g:RPM=0.0
BME280_Temp=0.0

B
dt [z]

W

{m}
Fan-PWM-Value{m}%PWM% Digit Fan-RPM{m}%RPM% U/m SDP810 DPressure{m}%DP% Pa
{m}
S
dt [l1c1]DP= %DP% Pa
dt [l2c1]RPM=%RPM% U/m
dt [l3c1]PWM=%PWM% Digit
dt [l4c1]PWM=%BME280_Temp% °C
###################################################################

How can I subscribe to a key and auto-update "in this example" the global variables?

MQTT-key Script-variable
"PWM" PWM
"DP_Pa" DP
"RPM" RPM
"Temperature" BME280_Temp

THANKS !!!!!!

**Comment by barbudor** [comment]
This discussion seems related to the same topic : https://github.com/arendst/Tasmota/discussions/16943
Apologies if not, I'm not a Script master

**Reply by HaraldNoack** [reply]
I'm also not a Script master :-)
So I have to ask and maybe it is helpful for other newbies to get a full example.
Thanks !
